receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet/kubelet.go

```diff
@@ -0,0 +1,206 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+// +build windows
+
+package kubelet // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows"
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
+	"go.uber.org/zap"
+	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+)
+
+// KubeletProvider Represents interface to kubelet.
+type KubeletProvider interface {
+	GetClient() (*kubeletutil.KubeletClient, error)
+	GetSummary() (*stats.Summary, error)
+}
+
+type kubeletProvider struct {
+	logger   *zap.Logger
+	hostIP   string
+	hostPort string
+	client   *kubeletutil.KubeletClient
+}
+
+// GetClient Returns singleton kubelet client.
+func (k *kubeletProvider) GetClient() (*kubeletutil.KubeletClient, error) {
+	if k.client != nil {
+		return k.client, nil
+	}
+	kclient, err := kubeletutil.NewKubeletClient(k.hostIP, k.hostPort, k.logger)
+	if err != nil {
+		k.logger.Error("failed to initialize kubeletProvider client, ", zap.Error(err))
```

Comment:
Can the error reporting be simplified by just returning the error?  No need to log boiler plate at each level

Reply:
this is existing pattern in containerinsights code.